lib/checknullpointer.cpp

```diff
@@ -279,47 +280,72 @@ static bool isNullablePointer(const Token* tok)
     return false;
 }
 
-void CheckNullPointer::nullPointerByDeRefAndChec()
+static bool isPointerUnevaluated(const Token *tok)
+{
+    if (isUnevaluated(tok))
```

Comment:
This should be checked in `findTokensSkipDeadCodeImpl` function.

Reply:
@danmar @pfultz2 I updated my PR to move this in `findTokensSkipDeadAndUnevaluatedCode` function because I was not very happy with the way I detected unevaluated pointers.